### PR TITLE
BUG: setting meta data type

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Follows #1095

When data is set to new values, the meta types are updated accordingly.  However, in some cleaning routines, values are updated to `np.nan`.  This breaks the type update.

~~This bugfix only updates the type if the value is non a nan.~~

The source of the bug is that `meta._data_types` is initiated as `None` by default.  When new data is set, keys cannot be added to a `NoneType`

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

By running pytest on pysatNASA.

**Test Configuration**:
* Operating system: Monterrey
* Version number: Python 3.9.7

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] Add a note to ``CHANGELOG.md``, summarizing the changes -- bugfix to last pull

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysat/_instrument.py [hunk 1]
```diff
@@ -1003,8 +1003,9 @@ def __setitem__(self, key, new_data):
                     # (including list or slice).
                     self.data.loc[self.data.index[key[0]], key[1]] = new
 
-                self.meta._data_types[key[1]] = self.data[
-                    key[1]].values.dtype.type
+                if not np.isnan(new):
+                    self.meta._data_types[key[1]] = self.data[
+                        key[1]].values.dtype.type
```
Comment (aburrell) [inline pysat/_instrument.py:1008]:
```suggestion
                if np.isnan(new):
                    self.meta._data_types[key[1]] = float
                else:
                    self.meta._data_types[key[1]] = self.data[
                        key[1]].values.dtype.type
```

Is this a better way to move forward?

**Review by aburrell — COMMENTED** [review]
A question on the approach.

File: pysat/_instrument.py (hunk 1 above)
Comment (jklenzing) [reply pysat/_instrument.py:1008]: I still get the same error when new is a nan:
```
TypeError: 'NoneType' object does not support item assignment
```

Comment (jklenzing) [reply pysat/_instrument.py:1008]: Actually, if I change the set function to 37.0 rather than nan, I still get the same error.

Comment (jklenzing) [reply pysat/_instrument.py:1008]: Found the source of the bug deeper in.  Updates pending the local tests...

**Comment by jklenzing** [comment]
The solution is more complex than I thought.  Closing in favor of writeup in #1109

**Pull request closed without merging**